src/BasePaintSubscription.sol

```diff
@@ -0,0 +1,105 @@
+// SPDX-License-Identifier: MIT
+
+//  ____                  ____       _       _
+// | __ )  __ _ ___  ___ |  _ \ __ _(_)_ __ | |_
+// |  _ \ / _` / __|/ _ \| |_) / _` | | '_ \| __|
+// | |_) | (_| \__ \  __/|  __/ (_| | | | | | |_
+// |____/ \__,_|___/\___||_|   \__,_|_|_| |_|\__|
+//  ____        _                  _       _   _
+// / ___| _   _| |__  ___  ___ _ __(_)_ __ | |_(_) ___  _ __
+// \___ \| | | | '_ \/ __|/ __| '__| | '_ \| __| |/ _ \| '_ \
+//  ___) | |_| | |_) \__ \ (__| |  | | |_) | |_| | (_) | | | |
+// |____/ \__,_|_.__/|___/\___|_|  |_| .__/ \__|_|\___/|_| |_|
+//                                   |_|
+
+pragma solidity ^0.8.17;
+
+import {OwnableUpgradeable} from "openzeppelin-contracts-upgradeable/contracts/access/OwnableUpgradeable.sol";
+import {IERC1155} from "openzeppelin-contracts/contracts/token/ERC1155/IERC1155.sol";
+import {ERC1155Upgradeable} from "openzeppelin-contracts-upgradeable/contracts/token/ERC1155/ERC1155Upgradeable.sol";
+import {Initializable} from "openzeppelin-contracts-upgradeable/contracts/proxy/utils/Initializable.sol";
+import {UUPSUpgradeable} from "openzeppelin-contracts-upgradeable/contracts/proxy/utils/UUPSUpgradeable.sol";
+
+interface IBasePaint is IERC1155 {
+    function openEditionPrice() external view returns (uint256);
+    function mint(uint256 day, uint256 count) external payable;
+    function today() external view returns (uint256);
+}
+
+contract BasePaintSubscription is Initializable, OwnableUpgradeable, ERC1155Upgradeable, UUPSUpgradeable {
+    IBasePaint public basepaint;
+    uint256 public discountPercentage;
+
+    error WrongEthAmount();
+    error InvalidSubscribedDay();
+    error InvalidDiscountPercentage();
+
+    struct Subscription {
+        uint256 day;
+        uint256 count;
+    }
+
+    function initialize(address _basepaint, address _owner) public initializer {
+        __Ownable_init(_owner);
+        __ERC1155_init("https://basepaint.xyz/api/subscription/{id}");
+        __UUPSUpgradeable_init();
+
+        basepaint = IBasePaint(_basepaint);
+        discountPercentage = 5;
+    }
+
+    function setDiscountPercentage(uint256 _discountPercentage) external onlyOwner {
+        if (_discountPercentage > 50) revert InvalidDiscountPercentage();
```

Comment:
doesn't make sense since it's what we control anyways, and we can upgrade the contract too